### DISCUSSION TITLE
Test executable not being made.

### DISCUSSION THREAD

**Discussion by Mobinati** [Chaste/Chaste#285] [Q&A] [answered]
Hello there, 
I am trying to run a project. under the folder of project I have myproject with the files I gonna send the zip file of them all. this is the tree 

├── CMakeLists.txt
├── src/
│   ├── MySimulation.cpp
│   ├── MySpvForce.cpp
│   └── MySpvForce.hpp
└── test/
    ├── CMakeLists.txt
    ├── TestMySpvForce.cpp
    └── TestMySimulation.hpp

I build the project in chaste_build out of chaste folder. everything gets built but my test executables in test folder. other projects test executables get built fine; just not the one I coded.
[myproject.zip](https://github.com/user-attachments/files/15747674/myproject.zip)

**Comment by mirams** [comment, marked as answer]
Hi @Mobinati, 
There's a bit of chaste-specific magic here. 

To save you work writing cmake commands to build those tests and add them to the ctest framework, we use various "test packs", to tell `cmake` and `ctest` they exist and let them be run easily. 

The default pack is defined in a plain text file `ContinuousTestPack.txt` within the `test` folder (as per https://github.com/Chaste/template_project/blob/main/test/ContinuousTestPack.txt for the template project), so you need that to be present in your project test folder and for it to list the filenames of your two tests. 

Then they should get configured to be built when you re-run `cmake`, built when you type `make`, and run when you do:
* e.g. `ctest -V -R TestMySpvForce` for an individual test; or
* `ctest -L myproject --output-on-failure` will run all the tests in your project.

Let me know if that sorts things out, does appear the documentation of this could be improved on the website!
